### PR TITLE
bt-268: Authorization plugin. Improving error return.

### LINKED ISSUE #268: Authorization plugin. Improving error return.
Authorization plugin returns similar error if there is no token and no authorized user is found.

Changed to: 
```
if (!token) {
  throw new HttpError({
    message: ErrorMessages.UNAUTHORIZED_USER,
    status: HttpCode.UNAUTHORIZED,
  });
}
........
if (!authorizedUser) {
  throw new HttpError({
    message: 'Invalid Token',
    status: HttpCode.UNAUTHORIZED,
  });
}
```

### PR DESCRIPTION
**Pull request by Monyster**
Closes #268

### CODE REVIEW

**Review by JosuerBague — APPROVED** [review]

**Review by MaksymGavatuk — APPROVED** [review]

File: backend/src/common/plugins/authorization/authorization.plugin.ts [hunk 1]
```diff
@@ -35,15 +36,21 @@ const authorizationPlugin: FastifyPluginCallback<AuthOptions> = (
         const [, token] = authorization?.split(' ') ?? [];
 
         if (!token) {
-            throw new Error(ErrorMessages.NOT_AUTHORIZED);
+            throw new HttpError({
+                message: ErrorMessages.MISSING_TOKEN,
```
Comment (nikita-remeslov) [inline backend/src/common/plugins/authorization/authorization.plugin.ts]: Unauthorized user

File: shared/src/enums/error-messages.enum.ts
Comment (nikita-remeslov) [inline shared/src/enums/error-messages.enum.ts:1]: we must keep errors universal so that later we can validate them more easily if necessary

**Review by nikita-remeslov — CHANGES_REQUESTED** [review]

File: backend/src/common/plugins/authorization/authorization.plugin.ts (hunk 1 above)
Comment (Monyster) [reply backend/src/common/plugins/authorization/authorization.plugin.ts]: You mean something like this?
```js
 if (!token) {
    throw new HttpError({
        message: ErrorMessages.UNAUTHORIZED_USER,     <--- "User not authorized"
        status: HttpCode.UNAUTHORIZED,       <--- 401
    });
}
```

File: shared/src/enums/error-messages.enum.ts [hunk 2]
```diff
@@ -4,6 +4,9 @@ const ErrorMessages = {
     INCORRECT_EMAIL: 'Incorrect email.',
     EMAIL_ALREADY_EXISTS: 'Email is already taken.',
     PASSWORDS_NOT_MATCH: 'Passwords do not match.',
+    MISSING_TOKEN: 'Missing token.',
```
Comment (nikita-remeslov) [inline shared/src/enums/error-messages.enum.ts]:
```suggestion
```

**Review by nikita-remeslov — CHANGES_REQUESTED** [review]

**Review by nikita-remeslov — APPROVED** [review]

**Pull request merged**